arviz/tests/base_tests/test_plots_matplotlib.py

```diff
@@ -951,6 +951,14 @@ def test_plot_posterior(models, kwargs):
         assert axes.shape
 
 
+def test_plot_posterior_boolean():
+    data = np.random.choice(a=[False, True], size=(4, 100))
+    axes = plot_posterior(data)
+    plt.draw()
```

Comment:
```suggestion
```

Reply:
I think this is not needed for tests (or is it?)

Reply:
I have an answer to this which may look weird at first as it was to me.
I was running an example on jupyter notebook. 
Calling `axes.get_xticklabels()` in the same cell `plot()` is called returns empty list.
However, I get the desired output when run `axes.get_xticklabels()` in next cell. 
I found a solution [here](https://github.com/matplotlib/matplotlib/issues/6103/) which suggested to add `plt.draw()`

Reply:
Answer to this question may look weird at first but
On notebook, assert axes.get_xticklabels() runs fine when run in next cell but returns empty list when run on the same cell containing `plt.plot()`.
Found a solution [here](https://github.com/matplotlib/matplotlib/issues/6103/)


(My earlier comment got duplicated. When I deleted one, both got deleted 🤦🏼 )

Reply:
Ok, then this can be removed (we don't run tests in notebooks)

Reply:
Tests were failing when I removed this because then get_xticklabels() returns empty list.

Interactive users when want to run `axes.get_xticklabels()` can do this by running it in different cell

But non interactive users have to call plt.draw() before using `axes.get_xticklabels()` 

More clear explanation : 
> in interactive mode in ipython, a draw() is automatically triggered after each function that has a plotting action. In non-interactive environments, the draw action is executed only at the end--this saves a lot of run time.


But maybe, we can call plt.draw() inside our posterior plot function and remove it from here.

Reply:
Yes, tests are really run from command line 

    python -m pytest path/to/tests

So keeping them clean is ok.